keras_contrib/optimizers/padam.py

```diff
@@ -0,0 +1,94 @@
+from keras import backend as K
+from keras.optimizers import Optimizer
+
+
+class Padam(Optimizer):
+    def __init__(self, lr=1e-1, beta_1=0.9, beta_2=0.999,
+                 epsilon=1e-8, decay=0., amsgrad=False, partial=1. / 8., **kwargs):
+        """ Partially adaptive momentum estimation optimizer.
+
+        # Arguments
+            lr: float >= 0. Learning rate.
+            beta_1: float, 0 < beta < 1. Generally close to 1.
+            beta_2: float, 0 < beta < 1. Generally close to 1.
+            epsilon: float >= 0. Fuzz factor. If `None`, defaults to `K.epsilon()`.
+            decay: float >= 0. Learning rate decay over each update.
+            amsgrad: boolean. Whether to apply the AMSGrad variant of this
+                algorithm from the paper "On the Convergence of Adam and
+                Beyond".
+            partial: float >=0. Parameter controlling partial momentum adaption.
+
+        # References
+            - [Closing the Generalization Gap of Adaptive Gradient Methods in Training Deep Neural Networks](https://arxiv.org/pdf/1806.06763.pdf)
+
+        """
+        super(Padam, self).__init__(**kwargs)
```

Comment:
Add a line to check that `partial` lies between 0 and 0.5, else raise a ValueError stating that it must be a positive float that can have a maximum value of 0.5, since anything higher will cause divergence during training. 

Better to have a fail fast behavior for something as pivotal as an optimizer.